cc-snapshot

```diff
@@ -1,13 +1,42 @@
 #!/bin/bash
 
-set -eu
-set -o pipefail
-
 if [[ $EUID -ne 0 ]]; then
    echo "This script must be run as root"
    exit 1
 fi
 
+###### Update section ######
+
+#Download latest version, assign versions to variables, curlstat variable ensures update only executes if curl succeeds.
+#------------------------------------------------
+SCRIPT=$0
+curl -s -o /tmp/tmp-snap https://raw.githubusercontent.com/ChameleonCloud/cc-snapshot/fhlb_test/cc-snapshot
+CURLSTAT=$?
+NEWSCRIPT='/tmp/tmp-snap'
+
+#Compare files and update if cc-snapshot outdated
+#------------------------------------------------
+diff -q $SCRIPT $NEWSCRIPT &>/dev/null
+DIFF=$?
+if [ $DIFF -eq 1 ] && [ $CURLSTAT -eq 0 ];
+then
+        echo 'Your cc-snapshot is not up to date.'
+        read -ep 'Update now? [y/n]'
+        if [[ $REPLY =~ ^[Yy]$ ]];
+        then
+                mv --backup=t $SCRIPT $SCRIPT'-old'
+                mv $NEWSCRIPT $SCRIPT
+                chmod +x $SCRIPT
+                echo 'Update complete. Run cc-snapshot again.'
```

Comment:
Just wondering, what would happen if this did a `exec "$SCRIPT"; exit $?` at the end? It's weird, but I think it would work.

Reply:
@diurnalist I think the `exit` is not needed in this case: see my most recent comment to @francoishalbach.